dwave/optimization/src/nodes/indexing.cpp

```diff
@@ -816,16 +814,15 @@ void AdvancedIndexingNode::propagate(State& state) const {
             std::pair<ssize_t, ssize_t> mapped_index =
                     get_mapped_index(indices_, array_strides, array_item_strides(),
                                      array_ptr_->shape(state), strides(), itemsize(), update.index);
-            if (!offsets_updated.contains(mapped_index.first)) {
-                const auto* idxs = node_data->get_offset_idxs(mapped_index.first);
-                if (idxs != nullptr) {
-                    for (const auto& idx : *idxs) {
-                        ssize_t item_stride = subspace_stride_ / itemsize();
-                        ssize_t new_index = idx * item_stride + mapped_index.second;
-                        assert(data[new_index] == update.old);
-                        diff.emplace_back(new_index, update.old, update.value);
-                        data[new_index] = update.value;
-                    }
+            const auto* idxs = node_data->get_offset_idxs(mapped_index.first);
+            if (idxs != nullptr) {
+                for (const auto& idx : *idxs) {
```

Comment:
NIT: I know this is what was there before. But I find `auto` in this context less helpful than just stating the type (I think `ssize_t`?). Also could add a couple `const` below. Happy enough to leave it alone as well.

Reply:
I agree, happy to change it. What are you looking at to add `const` to though?

Reply:
`item_stride`, `new_index` etc.